src/app/shared/disabled-directive.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  Directive,
+  HostBinding,
+  HostListener,
+  Input,
+} from '@angular/core';
+
+@Directive({
+  selector: '[dsDisabled]',
+  standalone: true,
+})
+export class DisabledDirective {
```

Comment:
Please add comments to this class to describe the purpose of this directive & how to use it.